### PR TITLE
FreeBSD fixes for pkg/mount

### PR DESCRIPTION
**Pull request by dfr**
This uses MNT_FORCE as an approximation of Linux MNT_DETACH and passes through key/value options such as devfs' ruleset=N option to the kernel. Also, I removed the code which truncates the zfs mount point to work around a path name limit in obsolete versions of FreeBSD which was fixed in FreeBSD 12.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

**Comment by flouthoc** [comment]
@dfr Could you rebase please although its not needed but github UI blocks merge button and only option is force merge.

**Comment by dfr** [comment]
Rebased

**Comment by rhatdan** [comment]
Ignoring the flakes.

**Pull request merged**
